biggraphite/drivers/cassandra.py

```diff
@@ -1456,17 +1456,37 @@ def __query_key(self, statement, params):
             else:
                 return str(statement) + str(params)
 
+    def _get_metrics(self, metric_names):
+        metric_names = [".".join(self._components_from_name(metric_name)[:-1])
+                        for metric_name in metric_names]
+        metric_names = [bg_metric.encode_metric_name(metric_name) for metric_name in metric_names]
+
+        statements_with_params = [self._make_select_metric_statement(metric_name)
+                                  for metric_name in metric_names]
+
+        results = self._select_metrics(statements_with_params)
+        rows = [result[0] for (success, result) in results if success and result[0] is not None]
```

Comment:
might be worth not putting that as a list comprehension, I think it would be more readable..
In the same for loop you could do this and the _bind_metric(), this would have the advantage of starting bind_metric() earlier (which is good because it also calls Cassandra with a self.has_directory())

Also I don't know if an exception should be sent when success is False here.

Reply:
* for comprehension: done
* about the exception, I'm wondering the same thing. I usually prefer returning no result rather that returning a wrong result. Here, the result would not be wrong but incomplete. Let's throw an exception and see if that is a problem when it (eventually) happens?